src/DynamicData/Cache/Internal/UniquenessEnforcer.cs

```diff
@@ -11,20 +11,45 @@ internal sealed class UniquenessEnforcer<TObject, TKey>(IObservable<IChangeSet<T
     where TKey : notnull
 {
     public IObservable<IChangeSet<TObject, TKey>> Run() =>
-        /*
-* If we handle refreshes, we cannot use .Last() as the last in the groupd may be a refresh,
-* and a previous in the group may add or update. Suddenly this scenario becomes very complicated
-* so for this phase we'll ignore these.
-*
+/*
+    For refresh,  we need to check whether there was a previous add or update in the batch. If not use refresh,
+    otherwise use the previous update.
 */
 
         source
```

Comment:
For virtualize I had to ensure unique changes, which previously did not work for Refresh